akka-http/src/main/scala/akka/http/scaladsl/server/directives/ReverseProxyDirectives.scala

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server.directives
+
+import scala.concurrent.Future
+import scala.util.Success
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.Uri.Authority
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers._
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.server.directives.ReverseProxyDirectives.{ ReverseProxyTargetConfig, ReverseProxyTargetMagnet }
+import akka.util.ByteString
+
+trait ReverseProxyDirectives {
+
+  def reverseProxy(target: ReverseProxyTargetMagnet): Route =
```

Comment:
By now, we'd rather not use magnets if it can be avoided. It seems here it is required to access the `system` to get the client. Maybe with the Akka 2.6 changes to use system more than materializer in many places, we could try to add a system to the `RequestContext` in the first place? That would be a follow up issue to this one.

Reply:
Are you suggesting we keep the magnet for now and remove it in a follow up issue, 
or remove it now and use overloaded functions that take an implicit system (removing the implicit parameter in the follow up issue)?

I support either course of action.